resources/idlharness.js

```diff
@@ -3437,16 +3437,52 @@ IdlNamespace.prototype.test_member_attribute = function (member)
     }.bind(this));
 };
 
+IdlNamespace.prototype.test_self = function ()
+{
+    const namespaceObject = self[this.name];
+
+    subsetTestByKey(this.name, test, () => {
+        const desc = Object.getOwnPropertyDescriptor(self, this.name);
+        assert_equals(desc.value, namespaceObject, `wrong value for ${this.name} namespace object`);
+        assert_true(desc.writable, "namespace object should be writable");
+        assert_false(desc.enumerable, "namespace object should not be enumerable");
+        assert_true(desc.configurable, "namespace object should be configurable");
+        assert_false("get" in desc, "namespace object should not have a getter");
+        assert_false("set" in desc, "namespace object should not have a setter");
+    }, `${this.name} namespace: property descriptor`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_true(Object.isExtensible(namespaceObject));
+    }, `${this.name} namespace: [[Extensible]] is true`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_true(namespaceObject instanceof Object);
+        assert_equals(Object.getPrototypeOf(namespaceObject), Object.prototype);
+    }, `${this.name} namespace: [[Prototype]] is Object.prototype`);
+
+    subsetTestByKey(this.name, test, () => {
+        assert_equals(typeof namespaceObject, "object");
+        assert_true(typeof namespaceObject === "object");
+        assert_false(typeof namespaceObject === "function");
+    }, `${this.name} namespace: typeof is "object"`);
+};
+
 IdlNamespace.prototype.test = function ()
 {
     /**
      * TODO(lukebjerring): Assert:
      * - "Note that unlike interfaces or dictionaries, namespaces do not create types."
-     * - "Of the extended attributes defined in this specification, only the
-     *     [Exposed] and [SecureContext] extended attributes are applicable to namespaces."
-     * - "Namespaces must be annotated with the [Exposed] extended attribute."
      */
 
+    subsetTestByKey(this.name, test, () => {
+        assert_true(this.extAttrs.every(o => o.name === "Exposed" || o.name === "SecureContext"),
+            "Only the [Exposed] and [SecureContext] extended attributes are applicable to namespaces");
+        assert_true(this.has_extended_attribute("Exposed"),
+            "Namespaces must be annotated with the [Exposed] extended attribute");
+    }, `${this.name} namespace: extended attributes`);
+
+    this.test_self();
```

Comment:
Should this be guarded by `if (!this.untested)`, like for `IdlInterface`? I'm not really familiar with the innards of idlharness...

Reply:
Seems like it should, added a guard in fb9484b.